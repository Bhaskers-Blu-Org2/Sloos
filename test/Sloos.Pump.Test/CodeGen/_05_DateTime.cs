// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT license.

using ApprovalTests.Reporters;
using Sloos.Pump.EntityFramework;
using Sloos.Pump.Test.EntityFramework;
using Sloos.Pump.Test.Samples;
using Xunit;

namespace Sloos.Pump.Test
{
    [UseReporter(typeof(DiffReporter))]
    public class _05_DateTime
    {
        [Fact]
        public void Reverse()
        {
            var entityNamespace = new EntityNamespace("Fact");

            var sampleGeneratorFactory = new SampleGeneratorFactory(
                CodeFirstGen.DateTime.Csdl(),
                CodeFirstGen.DateTime.Ssdl(),
                CodeFirstGen.DateTime.Msdl());

            var factory = new ReverseFactory(
                sampleGeneratorFactory,
                entityNamespace,
                "Context");

            ApprovalsExtensions.Verify(factory);
        }
    }
}
