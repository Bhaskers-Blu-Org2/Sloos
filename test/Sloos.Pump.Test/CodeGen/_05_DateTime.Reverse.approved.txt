namespace Fact.Models
{
    public partial class Context : global::System.Data.Entity.DbContext
    {
        static Context()
        {
            global::System.Data.Entity.Database.SetInitializer<Context>(null);
        }

        public Context(string nameOrConnectionString)
            : base(nameOrConnectionString)
        {
        }

        public global::System.Data.Entity.DbSet<global::Fact.Models.Table> Tables { get; set; }

        protected override void OnModelCreating(global::System.Data.Entity.DbModelBuilder modelBuilder)
        {
            modelBuilder.Configurations.Add(new global::Fact.Models.Mapping.TableMap());
        }
    }
}


namespace Fact.Models
{
    public partial class Table
    {
        public global::System.Int32 ID { get; set; }
        public global::System.DateTime LastUpdated { get; set; }
        public global::System.DateTime TypeIsDateTime1 { get; set; }
        public global::System.DateTime TypeIsDateTime2 { get; set; }
        public global::System.TimeSpan Time { get; set; }
        public global::System.TimeSpan Time_Precision0 { get; set; }
        public global::System.Byte[] TimeStamp { get; set; }
    }
}


namespace Fact.Models.Mapping
{
    public class TableMap : global::System.Data.Entity.ModelConfiguration.EntityTypeConfiguration<Table>
    {
        public TableMap()
        {
            // Primary Key
            this.HasKey(x => x.ID);

            // Properties
            this.Property(x => x.TimeStamp)
                .IsRequired()
                .IsFixedLength()
                .HasMaxLength(8)
                .IsRowVersion();

            // Table & Column Mappings
            this.ToTable("Tables", "dbo");
            this.Property(x => x.ID).HasColumnName("ID");
            this.Property(x => x.LastUpdated).HasColumnName("LastUpdated");
            this.Property(x => x.TypeIsDateTime1).HasColumnName("TypeIsDateTime1");
            this.Property(x => x.TypeIsDateTime2).HasColumnName("TypeIsDateTime2");
            this.Property(x => x.Time).HasColumnName("Time");
            this.Property(x => x.Time_Precision0).HasColumnName("Time_Precision0");
            this.Property(x => x.TimeStamp).HasColumnName("TimeStamp");

            // Relationships
        }
    }
}

